Utils/src/PDFWeightProducer.cc

```diff
@@ -175,14 +176,18 @@ void PDFWeightProducer::produce(edm::StreamID, edm::Event& iEvent, const edm::Ev
   if(genHandle.isValid()){
     auto helper = makeHelper(genHandle.product(),norm_);
     if((!found_scales and nScales_>0) or (!found_pdfs and nPDFs_>0)){
-      unsigned offset = 1;
       //renormalization/factorization scale weights
+      offset++;
```

Comment:
doesn't work for the more common case described in the previous comment. `offset` should be 0 when going into the `if(!found_pss...)` block, *unless* `found_scales` is true. in the current commit, `offset` will be (at least) 1 going into that block all the time. hence the suggestion to check the status of `found_scales` *inside* the `if(!found_pss...)` block when determining the offset.

Reply:
Hmm, I figured for the common case, we get both the PDFs and scales from the `LHEEventProduct` and subsequently, we would _never_ enter the block at L178 - thereby leaving `offset` at its initialized value of 0 upon going into the `if(!found_pss...)` block to get the PSs.
Conversely, for the stealth signal case, we'll go in at L178 and `offset` will come out as `nScales_+1` for use in the PS block. Sorry if I am missing something obvious here.

Reply:
Ah sorry, the case where it doesn't work is Pythia-only signals, where the only weights provided are PS weights in the `GenEventInfoProduct`. Those samples will still go into this block and increment the offset incorrectly.

Reply:
Gotcha, alright, how about [6eb19af](https://github.com/TreeMaker/TreeMaker/pull/670/commits/6eb19af2c1c89e55c1e9f085eb282a28427343d9) ?